### PR TITLE
#20 Create GitHub Actions automation

### PR DESCRIPTION
**Pull request by zganger**
This PR adds GitHub actions automation to address issue #20. The stages all pass excluding rubocop, but the complaints it lists are legitimate. I did not address them as I am not a ruby expert, but I would accept guidance and add a revision, or we can consider this working and publish a follow-up PR to address the rubocop concerns.

### CODE REVIEW

**Comment by mzagaja** [comment]
I'll try taking a look at this tomorrow. For the balance of complaints rubocop has do you want to run:

```
rubocop --auto-gen-config
```
and commit the result?

This should [create a todo config file](https://docs.rubocop.org/rubocop/configuration.html#automatically-generated-configuration) and allow CI to pass.

**Comment by zganger** [comment]
Thanks, I added the todo file in the latest commit.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+name: Test
+
+on:
+  push:
```
Comment (jonnycodesalot) [inline .github/workflows/test.yml:4]: I wonder if we would really want this, from the perspective of spending credits?  Might be adequate to just run it for PRs to main.  But having it on individual pushes on other branches is certainly handy if credits are effectively free.

Comment (jonnycodesalot) [reply .github/workflows/test.yml:4]: @mzagaja

**Review by jonnycodesalot — APPROVED** [review]
Looks good to me, look forward to leveraging it!

**Pull request merged**